app/components/contact-page-components/SendMessage.tsx

```diff
@@ -0,0 +1,84 @@
+"use client";
+
+import React, { useRef, useState } from "react";
+import emailjs from "@emailjs/browser";
+import { ZodError, z } from "zod";
+
+import SendMessageForm from "./SendMessageForm";
+
+interface SendMessageProps {
+  setShowSuccess: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const SendMessage: React.FC<SendMessageProps> = ({ setShowSuccess }) => {
+  const [errorList, setErrorList] = useState("");
+  const [formData, setFormData] = useState({
+    user_name: "",
+    user_email: "",
+    message: "",
+    contact_info: "",
+  });
+
+  const form = useRef<HTMLFormElement>(null);
+
+  const emailFormSchema = z.object({
+    user_name: z.string().nonempty("Name is required"),
+    user_email: z
+      .string()
+      .email("Invalid email format")
+      .nonempty("Email is required"),
+    message: z
+      .string()
+      .min(100, "Message should be at least 100 characters long"),
+    contact_info: z.string().nonempty("Contact information is required"),
+  });
+
+  const removeErrorMessage = (errorMsg: string) => {
+    setErrorList((prevErrors) => prevErrors.replace(errorMsg, "").trim());
+  };
+
+  const sendEmail = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    try {
+      emailFormSchema.parse({
+        user_name: e.currentTarget.user_name.value,
+        user_email: e.currentTarget.user_email.value,
+        message: e.currentTarget.message.value,
+        contact_info: e.currentTarget.contact_info.value,
+      });
+
+      if (form.current) {
+        emailjs.sendForm(
+          "service_5k4smcw",
+          "template_bsusg0x",
+          form.current,
+          process.env.NEXT_PUBLIC_EMAIL_JS_PUBLIC_KEY
+        );
+        setShowSuccess(true);
+      } else {
+        console.error("Form reference is null.");
+      }
+    } catch (error) {
+      if (error instanceof ZodError) {
+        const errorMessage = error.errors.map((err) => err.message).join("\n");
+        setErrorList(errorMessage);
+      } else {
+        console.error("An unknown error occurred:", error);
+      }
+    }
```

Comment:
It looks as if you're sending an email using email js. but not handling the promise it returns. It's good to know whether the email was sent successfully or if an error occurred. 

You may have assumed this would be caught in the catch block.  _The try...catch statement is indeed catching errors, but it's specifically designed to catch errors thrown by the Zod schema validation_ with emailFormSchema.parse(). If the schema validation fails, a ZodError is thrown, which your catch block is handling.

One approach using async/await would be as follows. You might need to tweak it a little. You can also use `.then`, but I personally prefer the simplicity and readability of using async/await.

```
if (form.current) {
      const response = await emailjs.sendForm(
        "service_5k4smcw",
        "template_bsusg0x",
        form.current,
        emailJsPublicKey
      );

      if (response.status === 200) {
        console.log("Email successfully sent!");
        setShowSuccess(true);
      } else {
        console.error(`Failed to send email: ${response.text}`);
      }
    } else {
      console.error("Form reference is null.");
    }
```

Reply:
thanks a lot. That's working way better now